tools/commands/integration_test.py

```diff
@@ -103,95 +147,192 @@ def _integration_test(plugin_dir, timeout):
+                    plugin_name, test_target,
+                    ['The runner cannot find any devices to run tests.']))
+            continue
+
+        match = re.search(_LOG_PATTERN, last_line.strip())
+        if not match:
+            test_results.append(
+                TestResult.fail(plugin_name, test_target,
+                                ['Log message is not formatted correctly.']))
+            continue
+
+        # In some cases, the command returns 0 for failed cases, so we check again
+        # with the last log message.
+        exit_code = process.returncode
+        if match.group(2) == 'All tests passed!':
+            exit_code = 0
+        elif match.group(2) == 'Some tests failed.':
+            errors.append(
+                'flutter-tizen test integration_test failed, see the output above for details.'
+            )
+            exit_code = 1
+
+        if exit_code == 0:
+            test_results.append(TestResult.success(plugin_name, test_target))
+        else:
+            test_results.append(
+                TestResult.fail(plugin_name, test_target, errors=errors))
+
+    return test_results
 
-    if exit_code == 0:
-        return TestResult.success()
-    else:
-        return TestResult.fail(errors)
+
+def _get_target_table():
+
+    def _parse_target_info(capability_info: str):
+        capability_info.rstrip()
+        profile_name = ''
+        platform_version = ''
+        for line in capability_info.split('\n'):
+            tokens = line.split(':')
+            if (tokens[0] == 'profile_name'):
+                profile_name = tokens[1]
+            elif (tokens[0] == 'platform_version'):
+                platform_version = tokens[1]
+        return profile_name, platform_version
+
+    completed_process = subprocess.run('sdb devices',
+                                       shell=True,
+                                       cwd='.',
+                                       universal_newlines=True,
+                                       stderr=subprocess.PIPE,
+                                       stdout=subprocess.PIPE)
+
+    table = {}
+    if completed_process.returncode == 0:
+        lines = completed_process.stdout.rstrip().split('\n')
+        for line in lines[1:]:
+            id = line.split(' ')[0]
+            completed_process = subprocess.run(f'sdb -s {id} capability',
+                                               shell=True,
+                                               cwd='.',
+                                               universal_newlines=True,
+                                               stderr=subprocess.PIPE,
+                                               stdout=subprocess.PIPE)
+            profile, platform_version = _parse_target_info(
+                completed_process.stdout)
+            if not profile or not platform_version:
+                print(f'''
+Cannot extract {profile} or {platform_version} information from device {id}
```

Comment:
Maybe you meant this.

```suggestion
Cannot extract profile or platform_version information from device {id}
```

Reply:
I totally did, thanks!